plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -26,6 +26,7 @@
 import io.trino.spi.type.TypeManager;
 
 import java.util.Map;
```

Comment:
> If catalogs are loaded concurrently, this can lead to situations where a connector accesses the configuration of another connector.

Can you add a test to this PR showcasing this behavior?

Reply:
The problem is that this is a non-deterministic bug. A test I came up with takes around 20 seconds because it requires many iterations to hit this scenario. I'll try to come up with a better approach.